pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+
+# elementary charge, from scipy.constants.value('elementary charge')
+elementary_charge_C = 1.6021766208e-19
+
+# Boltzmann constant, from scipy.constants.value('Boltzmann constant')
+boltzmann_J_per_K = 1.38064852e-23
+
+# Define standard test condition (STC) temperature in degrees Celsius
+T_stc_degC = 25.
+
+# Define standard test condition (STC) temperature in Kelvin
+T_stc_K = 273.15 + T_stc_degC
+
+
+def sum_current(resistance_shunt, resistance_series, nNsVth, current, voltage,
+                saturation_current, photocurrent):
+    """
+    Sum the current at the diode node of the single-diode model (SDM).
+
+    The current sum equaling zero is the fundamental invariant of the SDM, as
+    described in, e.g., De Soto et al. 2006 [1]. Reverse-voltage breakdown and
+    bypass diodes are not included. Ideal device parameters are specified by
+    resistance_shunt=numpy.inf and resistance_series=0.
+
+    Inputs to this function can include any broadcast-compatible combination
+    of scalar, numpy.ndarray, pandas.Series, and pandas.DataFrame, but it is
+    the caller's responsibility to ensure that the arguments all are within
+    their proper ranges.
+
+    Parameters
+    ----------
+    resistance_shunt : numeric
+        Shunt resistance in ohms under desired I-V curve conditions.
+        Often abbreviated ``Rsh``. ``0 < Rsh <= numpy.inf``
+
+    resistance_series : numeric
+        Series resistance in ohms under desired I-V curve conditions.
+        Often abbreviated ``Rs``. ``0 <= Rs < numpy.inf``
+
+    nNsVth : numeric
+        The product of three components. 1) The usual diode ideal factor
+        (``n``), 2) the number of cells in series (``Ns``), and 3) the cell
+        thermal voltage under the desired I-V curve conditions (``Vth``). The
+        thermal voltage of the cell (in volts) may be calculated as
+        ``k*temp_cell/q``, where ``k`` is Boltzmann's constant (J/K),
+        ``temp_cell`` is the temperature of the p-n junction in Kelvin, and
+        ``q`` is the charge of an electron (coulombs). ``0 < nNsVth``
+
+    current : numeric
+        The terminal current in amperes under desired I-V curve conditions.
+
+    voltage : numeric
+        The terminal voltage in volts under desired I-V curve conditions.
+
+    saturation_current : numeric
+        Diode saturation current in amperes under desired I-V curve
+        conditions. Often abbreviated ``I_0``. ``0 < I_0``
+
+    photocurrent : numeric
+        Light-generated current (photocurrent) in amperes under desired
+        I-V curve conditions. Often abbreviated ``I_L``. ``0 <= I_L``
+
+    Returns
+    -------
+    current_sum : numeric
+
+    References
+    ----------
+    [1] W. De Soto et al., "Improvement and validation of a model for
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88, 2006.
+    """
+
+    # Compute voltage at diode node first
+    voltage_diode = voltage + resistance_series * current
```

Comment:
If `voltage_diode` is used elsewhere, a function `voltage_diode(voltage, current, resistance_series)` would be useful.